app/models/CategorisationAnswers.scala

```diff
@@ -0,0 +1,145 @@
+import queries.RecordCategorisationsQuery
+
+final case class CategorisationAnswers(
+  assessmentValues: Seq[AssessmentAnswer],
+  supplementaryUnit: Option[Int]
+)
+
+object CategorisationAnswers {
+
+  implicit lazy val format: OFormat[CategorisationAnswers] = Json.format
+
+  private case class CategorisationDetails(index: Int, assessment: CategoryAssessment, answer: AssessmentAnswer)
+
+  def build(userAnswers: UserAnswers, recordId: String): EitherNec[ValidationError, CategorisationAnswers] =
+    (
+      buildAssessmentDetails(userAnswers, recordId),
+      getSupplementaryUnit(userAnswers, recordId)
+    ).parMapN(CategorisationAnswers.apply)
+
+  private def getSupplementaryUnit(userAnswers: UserAnswers, recordId: String) =
+    userAnswers.getOptionalPageValueForOptionalBooleanPage(
+      userAnswers,
+      HasSupplementaryUnitPage(recordId),
+      SupplementaryUnitPage(recordId)
+    )
+
+  private def buildAssessmentDetails(
+    userAnswers: UserAnswers,
+    recordId: String
+  ): Either[NonEmptyChain[ValidationError], Seq[AssessmentAnswer]] =
+    for {
+      recordCategorisations <- userAnswers.getPageValue(RecordCategorisationsQuery)
+      categorisationInfo    <- getCategorisationInfoForThisRecord(recordCategorisations, recordId)
+      answeredAssessments   <- getAssessmentsFromUserAnswers(categorisationInfo, userAnswers, recordId)
+      _                     <- ensureNoExemptionIsOnlyFinalAnswer(answeredAssessments, recordId)
+      _                     <- ensureHaveAnsweredTheRightAmount(answeredAssessments, categorisationInfo.categoryAssessments.size)
+    //  _                     <- ensureHaveNotSkippedAny(answeredAssessments)
+      justTheAnswers         = answeredAssessments.map(_.answer)
+    } yield justTheAnswers
+
+  private def getCategorisationInfoForThisRecord(recordCategorisations: RecordCategorisations, recordId: String) =
+    recordCategorisations.records
+      .get(recordId)
+      .map(Right(_))
+      .getOrElse(Left(NonEmptyChain.one(NoCategorisationDetailsForRecordId(RecordCategorisationsQuery, recordId))))
+
+  private def getAssessmentsFromUserAnswers(
+    categorisationInfo: CategorisationInfo,
+    userAnswers: UserAnswers,
+    recordId: String
+  ): EitherNec[ValidationError, Seq[CategorisationDetails]] = {
+    val answers = categorisationInfo.categoryAssessments.zipWithIndex
+      .map(assessment =>
+        (assessment._2, assessment._1, userAnswers.get(AssessmentPage(recordId, assessment._2)))
+      )
+      .filter(x => x._3.isDefined)
+      .map(x => CategorisationDetails(x._1, x._2, x._3.get))
+
+    if (answers.isEmpty) {
+      Left(NonEmptyChain(PageMissing(AssessmentPage(recordId, 1))))
+    } else {
+      Right(answers)
+    }
+  }
+
+  private def ensureNoExemptionIsOnlyFinalAnswer(
+    answeredAssessments: Seq[CategorisationDetails],
+    recordId: String
+  ): EitherNec[ValidationError, Done] = {
+
+    //Last answer can be a NoExemption. Others can't
```

Comment:
does this account for the fact that if they complete all the assessments (and get cat 3) then the last answer will not be NoExemption?

Reply:
Yup, its only looking at everything other than the last answer using the "answeredAssessments.reverse.tail"